tests/bson/bson-enum-002.phpt

```diff
@@ -0,0 +1,104 @@
+--TEST--
+BSON encoding: Enums implementing Unserializable
+--SKIPIF--
+<?php require __DIR__ . "/../utils/basic-skipif.inc"; ?>
+<?php skip_if_php_version('<', '8.1.0'); ?>
+--FILE--
+<?php
+
+require_once __DIR__ . '/../utils/basic.inc';
+
+enum MyEnum implements MongoDB\BSON\Unserializable
```

Comment:
Can you add a test that shows what happens when we provide a typeMap specifying an enum that doesn't implement `Unserializable`? IMO we shouldn't require the interface for enums as the method doesn't need to contain any logic.

Reply:
I can add that, but the exception is thrown in unrelated code (and I believe it was already tested -- for non enums, of course).

Reply:
Added a test in `bson-enum_error-003.phpt` (Typemap specifies enum that does not implement Unserializable).